### PR TITLE
Added events article

### PR DESCRIPTION
**Pull request by SimonBaeumer**

### CODE REVIEW

File: source/developers-guide/event-guide/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+An event system will be used to extend an application with your code or reacting on an action which was triggered by
+an user. For example you want to change the behaviour of a button, in that case you can use an event which will be
+fired every time the button was clicked. 
+
+## Example
+To get a better idea of how events are working we will take a look at a example.
+
+For our use case we will register on a pseudo event and try to cancel the application execution.
+
+### Definition of an event
```
Comment (tiagojsag) [inline source/developers-guide/event-guide/index.md]: This is a beginner's document. You are discussing advanced and unneeded topics (event registration) even before you introduced what is really vital and crucial here: event listening.

File: source/developers-guide/event-guide/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+`actionEvents` will be fired everytime an action will be called. For example if the `indexAction` in the `Account` controller
+was called the event `Enlight_Controller_Action_Frontend_Account_Index` will be fired.
+
+### Event names
+The event names of the `controller events` are generated dynamically. We will use in this list placeholders for the event
+names so you are able to build the event name you need by yourself.
+
+This list is also sorted according to the time in which the events are fired. 
+
+* `Enlight_Controller_Action_PreDispatch`
```
Comment (tiagojsag) [inline source/developers-guide/event-guide/index.md]: Move this up, and give examples of what MODULE, CONTROLLER and ACTION are. Also explain the difference between secure and non secure events

File: source/developers-guide/event-guide/index.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+## Application events
+`Application events` are events that are explicitly emitted in order to allow you certain modifications. In contrast to 
+the very generic global events, application events have certain use cases in mind. For example:
+* do not allow this customer to log in
+* do not allow this product to be put in the cart
+* modify the result of the current product query
+* invalidate the cache for a certain prodcut id
+* register a new command line tool
+
+### Notify events
```
Comment (tiagojsag) [inline source/developers-guide/event-guide/index.md]: ALL events fall into one of these 3 categories, but your guide does not make this clear. Please fix this

**Comment by tiagojsag** [comment]
Please replace your example
- It uses a fake event. It's not wrong, but it would be better if you could use a real, usable example
- Attach a demo plugin and use it as an example. See if the devdocs already have one that fits. Only use it if it is VERY SIMPLE. Remember: this is a very beginner's guide. If there isn't anything that fits create a plugin that listens to the login action and, if it's the user's birthday, adds a very simple frontend message saying "happy birthday". Remember that the key things are the event registration and keeping it simple.

File: source/developers-guide/event-guide/index.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+* Routing ended
+* Starting dispatch loop
+
+This list will give you an overview of all global lifecycle events in the order they will be triggered. 
+
+* `Enlight_Controller_Front_StartDispatch`
+* `Enlight_Controller_Router_Default`
+* `Enlight_Controller_Dispatcher_Default`
+* `Enlight_Controller_Request_RequestHttp`
+* `Enlight_Controller_Response_ResponseHttp`
```
Comment (hlohaus) [inline source/developers-guide/event-guide/index.md]: Request+Response-Events existieren nicht

File: source/developers-guide/event-guide/index.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+This list is also sorted according to the time in which the events are fired. 
+
+* `Enlight_Controller_Action_PreDispatch`
+* `Enlight_Controller_Action_PreDispatch_MODULE`
+* `Enlight_Controller_Action_PreDispatch_MODULE_CONTROLLER`
+* `Enlight_Controller_Action_MODULE_CONTROLLER_ACTION`
+* `Enlight_Controller_Action_PostDispatchSecure_MODULE_CONTROLLER`
+* `Enlight_Controller_Action_PostDispatchSecure_MODULE`
+* `Enlight_Controller_Action_PostDispatchSecure`
+* `Enlight_Controller_Action_PostDispatch_CONTROLLER`
```
Comment (hlohaus) [inline source/developers-guide/event-guide/index.md]: `Enlight_Controller_Action_PostDispatch_MODULE_CONTROLLER`

**Comment by tiagojsag** [comment]
@hlohaus this was going to be declined, but please confirm with bcremer before spending more time fixing it

**Comment by bcremer** [comment]
@hlohaus @tiagojsag Mr @dnoegel is currently rewriting the whole article from scratch.

Also: hi @tiagojsag :wink:

**Comment by tiagojsag** [comment]
Alrighty, let me know if you need a proof read or something. My life of
doing nothing all day in barcelona is boring (NNOOOTTTTT)

Anyway, hi everyone ;)

**Pull request closed without merging**